File: Boletines/General/pruebascpp/src/nodoA.cpp
#include <ros/ros.h>
#include <pruebascpp/Mensaje.h>

#include <iostream>
using namespace std;

int main(int argc, char ** argv){

    //Creamos el nodo principal no
    ros::init(argc, argv, "nodoA");
    ros::NodeHandle nh;

    //Creamos el publicador
    ros::Publisher pub = nh.advertise<pruebascpp::Mensaje>("topicA", 1000);

    //Creamos el mensaje
    pruebascpp::Mensaje msg;

    int contador = 0;

    //Creamos el bucle de publicacion
    ros::Rate loop_rate(1);

    while (ros::ok()) {
        contador += 1;
        msg.A = contador;
        pub.publish(msg);
        loop_rate.sleep();
    }
}

File: Boletines/General/pruebas2/src/nodoA.py
#!/usr/bin/env python3
import rospy 
from pruebas2.msg import Mensaje


#Creamos el nodo principal 
rospy.init_node("NodeA")

#Creamos un publicador
pub = rospy.Publisher("TopicA", Mensaje, queue_size=10)

#Creamos un objeto de tipo Mensaje
msg = Mensaje()

contador = 0
rospy.sleep(1)

#Creamos un bucle para enviar mensajes 
while not rospy.is_shutdown():
    contador += 1
    msg.A = contador
    
    print ("Publicando mensaje: ", msg.A)
    
    pub.publish(msg)
    
    rospy.sleep(1)
